### PR TITLE
Add Eq, Ord, Show instances for PasswordHash and Salt.

### PR DESCRIPTION
**Pull request by a-02**

### CODE REVIEW

File: sel/src/Sel/Hashing/Password.hs [hunk 1]
```diff
@@ -181,6 +196,7 @@ passwordHashToHexText = Base16.encodeBase16 . passwordHashToBinary
 --
 -- @since 0.0.1.0
 newtype Salt = Salt StrictByteString
+  deriving (Eq, Ord, Show)
```
Comment (kozross) [inline sel/src/Sel/Hashing/Password.hs:199]: It would be better for this to use explicit strategies.

**Review by kozross — CHANGES_REQUESTED** [review]

File: sel/src/Sel/Hashing/Password.hs [hunk 2]
```diff
@@ -74,6 +76,19 @@ import LibSodium.Bindings.Random
...
+      foreignPtrEq ph1 ph2 cryptoPWHashStrBytes
+
+instance Ord PasswordHash where
+  (PasswordHash ph1) `compare` (PasswordHash ph2) =
+    unsafeDupablePerformIO $
+      foreignPtrOrd ph1 ph2 cryptoPWHashStrBytes
+
+instance Show PasswordHash where
+  show (PasswordHash fptr) = foreignPtrShow fptr cryptoPWHashStrBytes
+
```
Comment (Kleidukos) [inline sel/src/Sel/Hashing/Password.hs:91]: If you could add:

```
-- |
--
-- @since 0.0.1.0
```

to the instances that would be perfect

**Review by Kleidukos — CHANGES_REQUESTED** [review]

File: sel/src/Sel/Hashing/Password.hs [hunk 3]
```diff
@@ -182,6 +200,18 @@ passwordHashToHexText = Base16.encodeBase16 . passwordHashToBinary
...
+instance Eq Salt where
+  (Salt s1) == (Salt s2) = s1 == s2
+
+-- | @since 0.0.1.0
+instance Ord Salt where
+  (Salt s1) `compare` (Salt s2) = s1 `compare` s2
+
+-- | @since 0.0.1.0
+instance Show Salt where
+  show (Salt s1) = show s1
```
Comment (Kleidukos) [inline sel/src/Sel/Hashing/Password.hs:213]: I believe there's been a misunderstanding, "explicit strategies" refers to this: https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/deriving_strategies.html

Comment (a-02) [reply sel/src/Sel/Hashing/Password.hs:213]: Ah. Sorry about that.

Comment (Kleidukos) [reply sel/src/Sel/Hashing/Password.hs:213]: no worries. :)

**Comment by a-02** [comment]
Everything's in, just can't get fourmolu to play nice locally.

**Comment by Kleidukos** [comment]
@a-02 what's the exact problem? Are you using fourmolu-0.12.0.0?

**Comment by a-02** [comment]
That might be it. I'm trying to lint things on my end but it keeps coming back as "all good".

**Comment by Kleidukos** [comment]
@a-02 I've tried something, let's see if it works

**Review by Kleidukos — APPROVED** [review]

**Comment by Kleidukos** [comment]
@kozross I think we're good for this PR :+1:

File: sel/sel.cabal [hunk 4]
```diff
@@ -33,7 +33,12 @@ common common
     -fprint-potential-instances -Wno-unticked-promoted-constructors
     -Werror=unused-imports
 
-  default-language: Haskell2010
+  default-language:   Haskell2010
+  default-extensions:
+    DerivingStrategies
+    GeneralizedNewtypeDeriving
+    ScopedTypeVariables
+    TypeApplications
```
Comment (kozross) [inline sel/sel.cabal:41]: Could we not do this please? Having extensions manually specified per-file is much clearer to a reader.

**Review by kozross — CHANGES_REQUESTED** [review]

File: sel/src/Sel/Hashing/Password.hs [hunk 5]
```diff
@@ -228,6 +240,7 @@ asciiByteStringToPasswordHash textualHash =
 --
 -- @since 0.0.1.0
 newtype Salt = Salt StrictByteString
+  deriving newtype (Eq, Ord, Show)
```
Comment (kozross) [inline sel/src/Sel/Hashing/Password.hs:243]: Each of these needs `@since`s too.

**Review by kozross — CHANGES_REQUESTED** [review]

**Review by kozross — APPROVED** [review]

**Pull request merged**